iothub/device/src/TransientFaultHandling/ExponentialBackoff.cs

```diff
@@ -98,9 +98,9 @@ public override ShouldRetry GetShouldRetry()
                 if (currentRetryCount < this.retryCount)
                 {
                     Random random = new Random();
-                    int num = (int)((Math.Pow(2.0, (double)currentRetryCount) - 1.0) * (double)random.Next((int)(this.deltaBackoff.TotalMilliseconds * 0.8), (int)(this.deltaBackoff.TotalMilliseconds * 1.2)));
-                    int num2 = (int)Math.Min(this.minBackoff.TotalMilliseconds + (double)num, this.maxBackoff.TotalMilliseconds);
-                    retryInterval = TimeSpan.FromMilliseconds((double)num2);
+                    double num = ((Math.Pow(2.0, currentRetryCount) - 1.0) * random.Next((int)(this.deltaBackoff.TotalMilliseconds * 0.8), (int)(this.deltaBackoff.TotalMilliseconds * 1.2)));
```

Comment:
1. Why do you need to use `double` which is significantly slower and harder to debug. The end result is a TimeSpan which is based on `milliseconds`. I doubt anyone would care of timings less than a millisecond. Instead you can use something like ulong which should be large enough for these purposes. You may also want to use a `checked` statement to ensure no overflows.

2. What does it mean here to have a timespan formed from Double.MinValue, Double.MaxValue or Infinity? That basically makes the device either not wait (spin-wait causing high CPU) or wait forever (hang).

Reply:
1) I'll make use of a checked block here and keep the numbers as integers then. If overflow occurs, the retry policy will just default to the max backoff.

2) This code does not create TimeSpans with Double.MinValue or Double.MaxValue. It creates TimeSpans within the min backoff and the max backoff